docs/navigation-advanced.md

```diff
@@ -0,0 +1,399 @@
+
+    public readonly object Parameter;
+
+    public readonly NavigationTransitionInfo InfoOverride;        
+
+    public NavigationConfig(bool registerOnBackStack = true, object parameter = null, NavigationTransitionInfo infoOverride = null)
+    {
+        RegisterOnBackStack = registerOnBackStack;
+        Parameter = parameter;
+        InfoOverride = infoOverride;            
+    }
+
+    public static NavigationConfig Default => new NavigationConfig();
+}
+```
+## Changes in project: 
+### Changes in ActivationService.cs
+- Initialize the main frame from `ActivateAsync()` replacing `Window.Current.Content = _shell?.Value ?? new Frame();` by 
+ ```csharp
+var frame = new Frame();
+if (_shell?.Value != null)
+{
+    frame.Content = _shell.Value;
+}
+NavigationService.InitializeMainFrame(frame);
+```
+
+- On the same method replace `NavigationService.Navigated += Frame_Navigated;` by `NavigationService.Navigated += OnNavigated;` and replace the event handler:
+
+ ```csharp
+ private void OnNavigated(object sender, NavigationArgs e)
+{
+    SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = NavigationService.CanGoBack ? AppViewBackButtonVisibility.Visible : AppViewBackButtonVisibility.Collapsed;
+}
+```
+### Changes in DefaultLaunchActivationHandler.cs
+-  Change HandleInternalAsync to 
+```csharp
+NavigationService.Navigate(_navElement, NavigationService.FrameKeyMain, new NavigationConfig(registerOnBackStack: true, parameter: args.Arguments));
+```
+
+- Change CanHandleInternal to 
+```csharp
+return !NavigationService.IsInitialized(NavigationService.FrameKeyMain);
+```
+
+## Using the advanced NavigationService
+
+### Navigate to startup page
+- On App.xaml.cs in CreateActivationService replace the existing code with `return new ActivationService(this, typeof(Views.StartUpPage));` to navigate directly to the StartupPage
+
+### Navigate from StartUpPage to ShellPage
+#### ShellPage changes:
+-  Add NavigationCacheMode="Required" to ShellPage.xaml.cs
+ - Initialize secondary frame on ShellPage adding the following code to Initialize method on ShellViewModel.cs or ShellPage.xaml.cs
+ ```csharp
+ NavigationService.InitializeFrame(NavigationService.FrameKeySecondary, shellFrame);
+ ```
+
+ - Subscribe to NavigationService.Navigated event, implement the eventHandler 
+ ```csharp
+ private void OnNavigated(object sender, NavigationArgs e)
+{
+    // Handle navigation only for navigations on secondary frame
+    if (e.FrameKey == NavigationService.FrameKeySecondary)
+    {
+        Selected = navigationView.MenuItems
+                    .OfType<NavigationViewItem>()
+                    .FirstOrDefault(menuItem => IsMenuItemForPageType(menuItem, e.SourcePageType));
+    }
+}
+```
+- Unregister the event handler when navigating away.
+
+#### On the StartupPage:
+
+ - Navigate to ShellPage (setting registerOnBackstack to false will cause the navigation not to be added to the backstack, so you won't be able to go back.
```

Comment:
This sounds like you're telling the page you're navigating to, to not be registered but actually it's removing the  page you're navigating from.  Can "registerOnBackstack" be better named?